lib/components/ButtonLink.jsx

```diff
@@ -0,0 +1,205 @@
+    fullWidth,
+    noPad
+  } = props
+
+  let defaultClasses =
+    'inline-block text-center leading-snug tracking-wide outline-none focus:outline-none active:outline-none font-bold '
+
+  if (fullWidth) {
+    defaultClasses += 'w-full'
+  } else {
+    defaultClasses += 'width-max-content'
+  }
+
+  const { backgroundClasses, borderClasses, textColorClasses } = getColorClasses(color, disabled)
+  paddingClasses = getPaddingClasses(paddingClasses, noPad)
+  roundedClasses = getRoundedClasses(roundedClasses)
+  textSizeClasses = getTextSizeClasses(textSizeClasses, size)
+  transitionClasses = getTransitionClasses(transitionClasses)
+  let cursorClasses = getCursorClasses(disabled)
+
+  return classnames(
+    backgroundClasses,
+    borderClasses,
+    textColorClasses,
+    defaultClasses,
+    paddingClasses,
+    roundedClasses,
+    size,
+    textSizeClasses,
+    transitionClasses,
+    cursorClasses,
+    className
+  )
+}
+
+const getColorClasses = (color, disabled) => {
+  if (disabled) {
+    return COLOR_CLASSES.disabled
+  }
+
+  switch (color) {
+    case 'primary': {
+      return COLOR_CLASSES.primary
+    }
+    case 'secondary': {
+      return COLOR_CLASSES.secondary
+    }
+    case 'tertiary': {
+      return COLOR_CLASSES.tertiary
+    }
+    case 'danger': {
+      return COLOR_CLASSES.danger
+    }
+    case 'warning': {
+      return COLOR_CLASSES.warning
+    }
+    case 'text_warning': {
+      return COLOR_CLASSES.text_warning
+    }
+    default: {
+      return COLOR_CLASSES.primary
+    }
+  }
+}
+
+const getCursorClasses = (disabled) => {
+  if (disabled) {
+    return 'cursor-not-allowed'
+  }
+  return 'cursor-pointer'
+}
+
+const getPaddingClasses = (paddingClasses, noPad) => {
+  if (noPad) {
+    return ''
+  }
+
+  if (paddingClasses) {
+    return paddingClasses
+  }
+
+  return 'py-1 px-6 sm:py-2 sm:px-10'
+}
+
+const getTextSizeClasses = (textSizeClasses, size) => {
+  if (textSizeClasses) {
+    return textSizeClasses
+  }
+
+  switch (size) {
+    case 'xs':
+      return `text-xxs sm:text-xs lg:text-sm`
+    case 'sm':
+      return `text-xs sm:text-sm lg:text-base`
+    case 'base':
+      return `text-sm sm:text-base lg:text-lg`
+    case 'lg':
+      return `text-base sm:text-lg lg:text-xl`
+    case 'xl':
+      return `text-lg sm:text-xl lg:text-2xl`
+    case '2xl':
+      return `text-xl sm:text-2xl lg:text-3xl`
+    case '3xl':
+      return `text-xl sm:text-3xl lg:text-4xl`
+    default:
+      return `text-sm sm:text-base lg:text-lg`
+  }
+}
+
+const getTransitionClasses = (transitionClasses) => {
+  return transitionClasses || 'trans trans-fast'
+}
+
+const getRoundedClasses = (roundedClasses) => {
+  return roundedClasses || 'rounded-full'
+}
+
+export function ButtonLink(props) {
+  let {
+    children,
+    as,
+    href,
+  } = props
+
+  const classes = getButtonClasses(props)
+
+  const linkProps = pick(props, [
+    'target',
+    'rel',
+  ])
+
+  return <Link
```

Comment:
We could add an `if (!as) {` case and only render the `<a>` so we can use this for external links as well

Reply:
That's what I just wrote about 😂